county_scrapers/alameda_county.py

```diff
@@ -0,0 +1,147 @@
+    race_keys = {"Latinx/Hispanic": "Hispanic/Latino", "Asian": "Asian", "African_Amer": "African_American/Black", 
+                 "White":"White", "Pacific_Islander": "Pacific_Islander", "Native_Amer": "Native_American", "Multiple_Race": "Multirace", 
+                 "Other": "Other_Race", "Unknown": "Unknown_Race"}
+    
+    # parse gender cases
+    for k,v in gender_keys.items():
+        out["case_totals"]["gender"][k] = demo_totals[v]
+    # parse race cases and deaths
+    for k, v in race_keys.items():
+        out["case_totals"]["race_eth"][k] = demo_totals[v]
+        out["death_totals"]["race_eth"][k] = demo_totals['Deaths\u2014' + v] # dashes are decoded as the escape sequence '\u2014'
+    # get age cases
+    out["case_totals"]["age_group"] = { k:v for k,v in demo_totals.items() if 'Age' in k}
+
+    with open('./county_data/alameda_county.json', 'w', encoding = 'utf-8') as f:
+        json.dump(out, f, ensure_ascii= False, indent=4)
+        
+    return json.dumps(out, indent=4) # for printing to console
+
+# Confirmed Cases and Deaths
+def get_timeseries(dataframe = False) -> Dict: 
+    """Fetch daily and cumulative cases and deaths by day
+    Returns a .json by default. If dataframe set to True, returns a 
+    pandas DataFrame."""
+
+    # query API
+    param_list = {'where':'0=0', 'resultType': 'none', 'outFields': 'Date,AC_Cases,AC_CumulCases,AC_Deaths,AC_CumulDeaths', 'outSR': 4326,'orderByField': 'Date', 'f': 'json'}
+    response = requests.get(cases_deaths, params=param_list)
+    parsed = json.loads(response.content)
+    features = [obj["attributes"] for obj in parsed['features']]
+    
+    # convert dates to 'yyyy/mm/dd'
+    for obj in features:
+        month, day, year = obj['Date'].split('/')
+        if int(month) < 10:
+            month = '0' + month
+        obj['Date'] = "{}/{}/{}".format(year, month, day)
+    
+    if not dataframe:
+        return features
+    
+    fields = parsed["fields"]
+    cols = [f['name'] for f in parsed['fields']]
+    dframe = pd.DataFrame(data=features, index=rows, columns=cols)
+    return dframe
+
+def get_cases_and_deaths(timeseries: pd.DataFrame): 
+    """Get latest date entry for cumulative cases and deaths. 
+    Since data was fetched to sort by date, the latest field will be the last entry. """    
+    return timeseries.tail(1)
+
+def get_demographics(dataframe = False) -> Dict:
+    """Fetch cases and deaths by age, gender, race, ethnicity
+    Returns a .json by default. If dataframe set to True, returns a pandas
+    DataFrame."""
+    param_list = {'where': "Geography='Alameda County'", 'outFields': '*', 'outSR':4326, 'f':'json'}
+    response = requests.get(demographics, params=param_list)
+    parsed = response.json()
+    fields = parsed['fields']
+    data = parsed['features'][0]['attributes']
+    #make a list of all datapoints with value '<10'
+    # due to some quirk in the json decoding, dashes are stored as the escape sequence '\u2014'
```

Comment:
FWIW, this isn’t a quirk in the decoding — the values just use em dashes (`U+2014`) instead of hyphens (which, typographically, makes sense, since they are being used to delimit a category and a thing, e.g. `Deaths—African_American/Black`). You can see them as em dashes on the open data site, too (e.g. https://data.acgov.org/datasets/AC-HCSA::alameda-county-covid-19-cases-case-rates-and-deaths-by-various-subgroups?selectedAttribute=Deaths_African_American_Black).

Not that this changes anything in practical terms here if what we want at the end of the day is a hyphen instead of an em dash. ¯\\\_(ツ)\_/¯